File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.ts
/**
 Do not return anything, modify matrix in-place instead.
 */
const modRowCols = function(matrix,row,col) {
    
    
    for(let i = 0; i<matrix.length; i++){
            
        for(let j = 0; j<matrix[i].length; j++){
            
            if(row == i || col == j)
                matrix[i][j] = 0;
        }
    }
    
}
function setZeroes(matrix: number[][]): void {
    
    let calrc = [];
    
    matrix.map((row,i)=>{
        
        row.map((col,j)=>{
            if(col == 0)
                calrc.push([i,j]);
        })
    })
    

    calrc.map(arr =>{
        modRowCols(matrix,arr[0],arr[1]);
       
    })
};

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.cpp
class Solution {
public:
    void conrowcol(vector<vector<int>>& matrix,int x,int y){
        for(int i = 0; i<matrix.size(); i++){
            for(int j =0; j<matrix[i].size(); j++){
                if(i == x || j == y)
                    matrix[i][j] = 0;
            }
        }
    }
    void setZeroes(vector<vector<int>>& matrix) {
        
        vector<vector<int>>v;
        for(int i = 0; i<matrix.size(); i++){
            for(int j = 0; j<matrix[i].size(); j++){
                if(matrix[i][j] == 0){
                    v.push_back({i,j});
                }
            }
        }
        
        for(auto i : v){
            conrowcol(matrix,i[0],i[1]);
        }
    }
};